### DISCUSSION TITLE
Implement Modeling Standards

### DISCUSSION THREAD

**Discussion by jondavid-black** [DevOps-MBSE/AaC#387] [Ideas]
Currently AaC uses a custom (and highly extensible) approach to modeling a system architecture.  While this may provide value to engineers who wish to use an architecture model to actually build a system (and are willing to learn a new modeling vocabulary), there are others who simply want the architecture artifact to facilitate documentation and communication.  For these stakeholders, alignment to standards is important.

I'd like to recommend we demonstrate AaC extensibility by creating a [SysML](https://sysml.org/) plugin that implements the SysML schema for modeling.  Initially, this would just be a schema representation but would evolve to enable diagram generation and export/import to/from commercial tool formats (i.e Cameo XML).

Another recommendation is to create a model plugin representing architectural framework standards such as [Unified Architecture Framework (UAF)](https://www.omg.org/uaf/).

The final recommendation is to create a model plugin representing business processes such as [Business Process Model and Notation (BPMN)](https://www.bpmn.org/).

Enabling users to work with these standards out of the box should reduce the learning curve for modelers by using vocabularies they know and overcoming potential obstacles to adoption within organizations.

**Comment by cvchaparro** [comment]
We like the idea of expanding our support of these standard tools but we'll need a bit more information to make sure we can provide the most valuable components of these plugins from the get-go.

To that end, what would be the MVP for these plugins?
  * To start with would we mainly target conversion between AaC models and SysML models, etc?
  * Which features of SysML do we want to support?
  * Which features of UAF do we want to support?

Regarding BPMN features you would want us to support:
  * For conversions between BPMN and AaC, do you have some thoughts on how to represent business process models in AaC?

**Comment by Coffee2Bits** [comment]
Here are some of the specs we found for reference.
https://sysml.org/.res/docs/specs/OMGSysML-v1.5-17-05-01.pdf
https://www.omg.org/spec/BPMN/2.0/PDF
https://www.omg.org/spec/UAF/1.2/Beta1/UAFML/PDF
https://www.omg.org/spec/UAF/1.2/Beta1/DMM/PDF
